### PR TITLE
Correctly redirect the users on continuous applications

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

This PR fixes the redirect bugs for continuous applications in the 2024 recruitment cycle.

When the user will reach the complete page they will be redirected to continuous applications if (2024 recruitment cycle and continuous applications feature flag enabled)

This PR also adds system specs for all the scenarios related to carry over logic of continuous applications.

### CODE REVIEW

**Comment by vassyz** [comment]
[CA: Fix redirects](https://trello.com/c/EYLi6Vyo/526-ca-fix-redirects)

**Review by inulty-dfe — APPROVED** [review]
Great test coverage :+1:

File: spec/system/candidate_interface/continuous_applications/submitting/redirect_to_your_applications_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  def given_i_have_in_progress_application_form
+    application_form
+  end
+
+  def then_i_should_be_on_your_details_page
+    expect(page).to have_current_path candidate_interface_continuous_applications_details_path
+  end
+
+  def application_form
+    @application_form = create(
```
[the remark this answers is not in the document]
Comment (vassyz) [reply spec/system/candidate_interface/continuous_applications/submitting/redirect_to_your_applications_spec.rb:58]: No, good spot!

[the remark this answers is not in the document]
Comment (vassyz) [reply spec/system/candidate_interface/continuous_applications/submitting/redirect_to_your_applications_spec.rb]: Removed the instance variable.

**Pull request merged**